samples/bluetooth/iso_broadcast/src/main.c

```diff
@@ -0,0 +1,178 @@
+};
+
+static K_SEM_DEFINE(sem_big_cmplt, 0, 1);
+static K_SEM_DEFINE(sem_big_term, 0, 1);
+
+static void iso_connected(struct bt_iso_chan *chan)
+{
+	printk("ISO Channel %p connected\n", chan);
+	k_sem_give(&sem_big_cmplt);
+}
+
+static void iso_disconnected(struct bt_iso_chan *chan, uint8_t reason)
+{
+	printk("ISO Channel %p disconnected with reason 0x%02x\n",
+	       chan, reason);
+	k_sem_give(&sem_big_term);
+}
+
+static struct bt_iso_chan_ops iso_ops = {
+	.connected	= iso_connected,
+	.disconnected	= iso_disconnected,
+};
+
+#define BIS_ISO_CHAN_COUNT 1
+
+static struct bt_iso_chan_io_qos iso_tx_qos = {
+	.interval = 10000, /* in microseconds */
+	.latency = 10, /* milliseconds */
+	.sdu = 502, /* bytes */
+	.rtn = 3,
+	.phy = BT_GAP_LE_PHY_2M,
+};
+
+static struct bt_iso_chan_qos bis_iso_qos = {
+	.tx = &iso_tx_qos,
+	.packing = 0, /* 0 - sequential, 1 - interleaved */
+	.framing = 0, /* 0 - unframed, 1 - framed */
+};
+
+static struct bt_iso_chan bis_iso_chan = {
+	.ops = &iso_ops,
+	.qos = &bis_iso_qos,
+};
+
+static struct bt_iso_chan *bis[BIS_ISO_CHAN_COUNT] = { &bis_iso_chan };
+
+static struct bt_iso_big_create_param big_create_param = {
+	.num_bis = BIS_ISO_CHAN_COUNT,
+	.bis_channels = bis,
+};
+
+void main(void)
+{
+	struct bt_le_ext_adv *adv;
+	struct bt_iso_big *big;
+	int err;
+
+	printk("Starting ISO Broadcast Demo\n");
+
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	/* Create a non-connectable non-scannable advertising set */
+	err = bt_le_ext_adv_create(BT_LE_EXT_ADV_NCONN_NAME, NULL, &adv);
+	if (err) {
+		printk("Failed to create advertising set (err %d)\n", err);
+		return;
+	}
+
+	/* Set periodic advertising parameters */
+	err = bt_le_per_adv_set_param(adv, BT_LE_PER_ADV_DEFAULT);
+	if (err) {
+		printk("Failed to set periodic advertising parameters"
+		       " (err %d)\n", err);
+		return;
+	}
+
+	/* Enable Periodic Advertising */
+	err = bt_le_per_adv_start(adv);
+	if (err) {
+		printk("Failed to enable periodic advertising (err %d)\n", err);
+		return;
+	}
+
+	/* Start extended advertising */
+	err = bt_le_ext_adv_start(adv, BT_LE_EXT_ADV_START_DEFAULT);
+	if (err) {
+		printk("Failed to start extended advertising (err %d)\n", err);
+		return;
+	}
+
+	/* Create BIG */
+	err = bt_iso_big_create(adv, &big_create_param, &big);
+	if (err) {
+		printk("Failed to create BIG (err %d)\n", err);
+		return;
+	}
+
+	printk("Waiting for BIG complete...");
+	err = k_sem_take(&sem_big_cmplt, K_FOREVER);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("done.\n");
+
+	while (true) {
+		static uint8_t timeout = BIG_TERMINATE_TIMEOUT;
+
+		k_sleep(K_SECONDS(1));
+
+		mfg_data[2]++;
+
+		printk("Set Periodic Advertising Data...");
+		err = bt_le_per_adv_set_data(adv, ad, ARRAY_SIZE(ad));
+		if (err) {
+			printk("failed (err %d)\n", err);
+			return;
+		}
+		printk("done.\n");
```

Comment:
Why does the broadcaster set periodic advertising data? 

I suppose it should broadcast ISO data instead using `bt_iso_chan_send` instead

Reply:
This is to test periodic advertising data changes while maintaining the BIGInfo. `bt_iso_chan_send` is not added to this sample yet. 

Feel free to include it here and send a commit to this PR (Zephyr controller does not have data path hooked yet to test myself the data).

Reply:
Hmm, I guess it makes sense to test that, but does that belong in a sample application? It shouldn't be necessary for the application to set any PA data, and thus I suggest we remove that, or at least, add a todo to remove it and once we have data path then remove it. 

Imo setting PA data doesn't belong in a ISO broadcast sample.